src/databricks/labs/dqx/llm/utils.py

```diff
@@ -0,0 +1,25 @@
+import inspect
+from databricks.labs.dqx.engine import DQEngineCore
+from databricks.labs.dqx.rule import CHECK_FUNC_REGISTRY
+
+
+def get_check_function_defintion() -> list[dict[str, str]]:
+    """A utility function to get the definition of all check functions.
+    This is function is primarily used to generate a prompt for the LLM to generate a check functions.
+
+    Returns:
+      list[dict]: A list of dictionaries, each containing the definition of a check function.
+    """
+    function_docs: list[dict[str, str]] = []
+    for name, func_type in CHECK_FUNC_REGISTRY.items():
+        func = DQEngineCore.resolve_check_function(name, fail_on_missing=True)
```

Comment:
we will also need to check which functions within CHECK_FUNC_REGISTRY are built-in (those belonging to the `check_funcs` module)  and which are custom funcs. Compare funcs from CHECK_FUNC_REGISTRY with those available in `check_funcs` module. Anything extra is a custom function. 

You will need to collect custom_check_functions and pass it to the `resolve_check_function `. For custom checks we will need to include full implementation as docstrings are not guraranteed.